Source/Dafny/Compilers/Compiler-Csharp.cs

```diff
@@ -124,6 +124,46 @@ public CsharpCompiler(ErrorReporter reporter)
           w.WriteLine("return a;");
         }
       }
+
+      //Generates casts for functions of those arities present in the program
```

Comment:
Could you put the block that the comment pertains to in a separate method?

Could you also show put an example of a generated extension function in a comment?